lib/fluent/supervisor.rb

```diff
@@ -295,42 +299,48 @@ def self.load_config(path, params = {})
     end
 
     class LoggerInitializer
-      def initialize(path, level, chuser, chgroup, opts)
+      def initialize(path, level, chuser, chgroup, log_rotate_age, log_rotate_size, opts)
         @path = path
         @level = level
         @chuser = chuser
         @chgroup = chgroup
+        @log_rotate_age = log_rotate_age
+        @log_rotate_size = log_rotate_size
         @opts = opts
       end
 
       def init
         if @path && @path != "-"
-          @io = File.open(@path, "a")
+          logdev = if @log_rotate_age || @log_rotate_size
+                     Fluent::LogDeviceIO.new(@path, shift_age: @log_rotate_age, shift_size: @log_rotate_size)
+                   else
+                     File.open(@path, "a")
+                   end
           if @chuser || @chgroup
             chuid = @chuser ? ServerEngine::Privilege.get_etc_passwd(@chuser).uid : nil
             chgid = @chgroup ? ServerEngine::Privilege.get_etc_group(@chgroup).gid : nil
             File.chown(chuid, chgid, @path)
           end
         else
-          @io = STDOUT
+          logdev = STDOUT
         end
 
         dl_opts = {}
         # subtract 1 to match serverengine daemon logger side logging severity.
         dl_opts[:log_level] = @level - 1
-        logger = ServerEngine::DaemonLogger.new(@io, dl_opts)
+        logger = ServerEngine::DaemonLogger.new(logdev, dl_opts)
         $log = Fluent::Log.new(logger, @opts)
         $log.enable_color(false) if @path
         $log.enable_debug if @level <= Fluent::Log::LEVEL_DEBUG
       end
 
       def stdout?
-        @io == STDOUT
+        $log.out == STDOUT
```

Comment:
Sorry. I forgot it. 
I updated `@io` to `@logdev`.  Because the name of `@io` instance variable  is changed here (https://github.com/fluent/fluentd/pull/1235/files#diff-dbe0e1ec4079138e48ca6a4d7c7248f9R316).